libbeat/common/kubernetes/k8skeystore/kubernetes_keystore.go

```diff
@@ -76,12 +78,9 @@ func (kr *KubernetesKeystoresRegistry) GetKeystore(event bus.Event) keystore.Key
 	}
 	if namespace != "" {
 		// either retrieve already stored keystore or create a new one for the namespace
-		if storedKeystore, ok := kr.kubernetesKeystores[namespace]; ok {
-			return storedKeystore
-		}
 		k8sKeystore, _ := Factoryk8s(namespace, kr.client, kr.logger)
-		kr.kubernetesKeystores["namespace"] = k8sKeystore
-		return k8sKeystore
+		storedKeystore, _ := kr.kubernetesKeystores.LoadOrStore(namespace, k8sKeystore)
+		return storedKeystore.(keystore.Keystore)
```

Comment:
This way we are instantiating a new keystore on every call, this is why I mentioned that we could need to store an object that creates the keystore lazily.

But, I have been looking at the implementation of the k8s keystore and it actually doesn't keep any state at all, so maybe we can return a new one each time and don't keep a registry?

```suggestion
		return k8sKeystore
```